### PR TITLE
Adding test to make sure HEAD can run on VM created by latest release

### LINKED ISSUE #816: Add integration test to assert that HEAD can use state created by latest release
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Feature request

The 0.12.0 to 0.12.1 release required a recreate of the minikube VM because of localkube DNS changes.  This caused a lot of confusion for users upgrading. (#781) 

Minikube should ideally be able to be upgraded without recreating the VM.  We should add a test during our release job to enforce this behavior.

In the rare case that we can't get around recreating the VM, we should make a clear note of it in the release docs.

### PR DESCRIPTION
**Pull request by sharifelgamal**
Closes #816

### CODE REVIEW

File: test/integration/version_upgrade_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	currentRunner.CheckStatus(state.None.String())
+
+	isoName := "minikube-linux-amd64"
+	if runtime.GOOS == "darwin" {
+		isoName = "minikube-darwin-amd64"
+	} else if runtime.GOOS == "windows" {
+		isoName = "minikube-windows-amd64.exe"
+	}
+	// Grab latest release binary
+	url := "https://storage.googleapis.com/minikube/releases/latest/" + isoName
```
Comment (tstromberg) [inline test/integration/version_upgrade_test.go]: See if util.GetBinaryDownloadURL() is applicable for your use case.

File: test/integration/version_upgrade_test.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	resp, err := http.Get(url)
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "Failed to get latest release binary"))
+	}
+	defer resp.Body.Close()
+
+	iso, err := ioutil.TempFile("", isoName)
```
Comment (tstromberg) [inline test/integration/version_upgrade_test.go]: nit: I'd just name this "tf". "iso" doesn't have any meaning to me.

File: test/integration/version_upgrade_test.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "Failed to create binary file"))
+	}
+	defer os.Remove(iso.Name())
+
+	_, err = io.Copy(iso, resp.Body)
```
Comment (tstromberg) [inline test/integration/version_upgrade_test.go]: non-actionable note: other parts of the minikube code base use github.com/jimmidyson/go-download, but I'm not really sure of the benefits of doing so.

File: test/integration/version_upgrade_test.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "Failed to populate iso file"))
+	}
+	if err := iso.Close(); err != nil {
+		t.Fatal(errors.Wrap(err, "Failed to close iso"))
+	}
+
+	if runtime.GOOS != "windows" {
+		if err := exec.Command("chmod", "+x", iso.Name()).Run(); err != nil {
```
Comment (tstromberg) [inline test/integration/version_upgrade_test.go]: Prefer native functions: use os.Chmod instead.

File: test/integration/version_upgrade_test.go [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+			t.Fatal(errors.Wrap(err, "Failed to make binary executable."))
+		}
+	}
+
+	latestRunner := util.MinikubeRunner{
```
Comment (tstromberg) [inline test/integration/version_upgrade_test.go]: "latest" and "current" is difficult to differentiate between. How about something like "releaseRunner"?

File: test/integration/version_upgrade_test.go (hunk 4 above)
Comment (sharifelgamal) [reply test/integration/version_upgrade_test.go]: Done.

File: test/integration/version_upgrade_test.go (hunk 5 above)
Comment (sharifelgamal) [reply test/integration/version_upgrade_test.go]: Done.

File: test/integration/version_upgrade_test.go (hunk 2 above)
Comment (sharifelgamal) [reply test/integration/version_upgrade_test.go]: Done.

File: test/integration/version_upgrade_test.go (hunk 1 above)
Comment (sharifelgamal) [reply test/integration/version_upgrade_test.go]: It most certainly is!

**Comment by tstromberg** [comment]
Thanks!

**Pull request merged**